### ISSUE TITLE
Performance of resnet101

### ISSUE THREAD

**Issue by kleinzcy** [saic-vul/fbrs_interactive_segmentation#25]
Hi, authors, thanks to your nice paper and code!

Recently, I retrain the resnet101 model on your code. But my result is not as good as reported in the paper. I have read the issue but did not find any helpful information.  

My environment: Ubuntu16.04, CUDA10.1, pytorch1.3.0, four TITAN XP GPU

My results(NoBRS, last checkpoints, NFL):
 
![image](https://user-images.githubusercontent.com/35723500/92251771-e9837880-eeff-11ea-9adf-cc0e2e9a23b8.png)

results after f-BRS-B:
![image](https://user-images.githubusercontent.com/35723500/92251908-1b94da80-ef00-11ea-8f67-9628d25a3cd9.png)

And, the training curve is strange:
![image](https://user-images.githubusercontent.com/35723500/92252099-5e56b280-ef00-11ea-9036-480dee2eaaf5.png)

![image](https://user-images.githubusercontent.com/35723500/92252235-89410680-ef00-11ea-9912-95f0ca20105f.png)

The training loss is growing or constant(change slightly). Do you have any idea?

Thanks.

**Comment by kleinzcy** [comment]
When I retrain the resnet101 model on GTX 1080 Ti, the result is as good as reported in the paper and even better.
![image](https://user-images.githubusercontent.com/35723500/93036020-34745d00-f671-11ea-901f-824e58a0f347.png)

![image](https://user-images.githubusercontent.com/35723500/93036030-3f2ef200-f671-11ea-84f5-8d4a1bda5cdb.png)

The NoC metric is sensitive.

**Comment by MaitaYuki** [comment]
@kleinzcy , hope you do not mind I ask you a question since I encountered the same problem you mentioned in the beginning: during training, the training and validation loss do not change much from epoch 1 to epoch 120, fluctuate around 0.3 as shown below, 

(INFO) 2021-02-09 06:50:46: Epoch 99, training loss 0.319589:  96%|########################## | 512/531 [05:30<00:12,  1.56it/s]
(INFO) 2021-02-09 06:50:51: Epoch 99, training loss 0.319661:  98%|##########################4| 520/531 [05:35<00:07,  1.56it/s]
(INFO) 2021-02-09 06:50:57: Epoch 99, training loss 0.319605:  99%|##########################8| 528/531 [05:40<00:01,  1.55it/s]
(INFO) 2021-02-09 06:50:58: Save checkpoint to experiments/sbd/r34_dh128/008_first-try/checkpoints/last_checkpoint.pth
(INFO) 2021-02-09 06:51:02: Epoch 99, validation loss: 0.339068:   7%|#6                       | 12/178 [00:03<00:32,  5.03it/s]
(INFO) 2021-02-09 06:51:07: Epoch 99, validation loss: 0.341119:  21%|#####3                   | 38/178 [00:08<00:27,  5.15it/s]
(INFO) 2021-02-09 06:51:12: Epoch 99, validation loss: 0.341862:  36%|########9                | 64/178 [00:13<00:22,  5.14it/s]

When you reported your result is as good as reported in the paper and even better,  did you observe the loss reduced below 0.3 a lot? How did you solve the problem? Thanks.

**Comment by kleinzcy** [comment]
@MaitaYuki Sorry for the late reply. I am on a long holiday.

Loss fluctuates around 0.3 because of normalized focal loss. You can look at its formulation.

**Comment by Looottch** [comment]
> When I retrain the resnet101 model on GTX 1080 Ti, the result is as good as reported in the paper and even better.
> ![image](https://user-images.githubusercontent.com/35723500/93036020-34745d00-f671-11ea-901f-824e58a0f347.png)
> 
> ![image](https://user-images.githubusercontent.com/35723500/93036030-3f2ef200-f671-11ea-84f5-8d4a1bda5cdb.png)
> 
> The NoC metric is sensitive.



> @kleinzcy , hope you do not mind I ask you a question since I encountered the same problem you mentioned in the beginning: during training, the training and validation loss do not change much from epoch 1 to epoch 120, fluctuate around 0.3 as shown below,
> 
> (INFO) 2021-02-09 06:50:46: Epoch 99, training loss 0.319589: 96%|########################## | 512/531 [05:30<00:12, 1.56it/s]
> (INFO) 2021-02-09 06:50:51: Epoch 99, training loss 0.319661: 98%|##########################4| 520/531 [05:35<00:07, 1.56it/s]
> (INFO) 2021-02-09 06:50:57: Epoch 99, training loss 0.319605: 99%|##########################8| 528/531 [05:40<00:01, 1.55it/s]
> (INFO) 2021-02-09 06:50:58: Save checkpoint to experiments/sbd/r34_dh128/008_first-try/checkpoints/last_checkpoint.pth
> (INFO) 2021-02-09 06:51:02: Epoch 99, validation loss: 0.339068: 7%|#6 | 12/178 [00:03<00:32, 5.03it/s]
> (INFO) 2021-02-09 06:51:07: Epoch 99, validation loss: 0.341119: 21%|#####3 | 38/178 [00:08<00:27, 5.15it/s]
> (INFO) 2021-02-09 06:51:12: Epoch 99, validation loss: 0.341862: 36%|########9 | 64/178 [00:13<00:22, 5.14it/s]
> 
> When you reported your result is as good as reported in the paper and even better, did you observe the loss reduced below 0.3 a lot? How did you solve the problem? Thanks.

Did you remenber whole time of retraining and your how many gpus you used

**Comment by Looottch** [comment]
@kleinzcy @MaitaYuki Did you remember how many time will 120 epoch takes (and how many gpus you used)
